docs/development/data-flow/README.md

```diff
@@ -0,0 +1,98 @@
+
+The external web server acts as a proxy/reverse-proxy for the OpenProject Puma application server, relaying requests for it to handle and respond. In the course of the request, access to external services such as the PostgreSQL database, a caching server, or attached storages might be performed.
+
+In the course of using the application, background tasks are enqueued in the database such as outgoing emails, cleanup tasks, or notification processing. These tasks are performed in a separate process, the background worker queue. This process accesses the same services as the application server process to access or modify data. It might connect to external integrations such as a Nextcloud instance to set up file sharings depending on actions performed by the users.
+
+
+
+## Schema information
+
+OpenProject's database schema is considered an internal API, please do not rely on it as a stable point of references. Modifications to the database schema are considered a third-party customization and might prevent us from giving proper support.
+
+We recommend you use existing database tools to inspect and export the database as an ERD. That being said, we've prepared an export of an ERD of the current state of the database. Newer additions to it might be missing.
+
+[![OpenProject database schema ERD](openproject-erd.jpg)](openproject-erd.jpg)
+
+
+
+# Use of technical cookies
+
+OpenProject makes use of technical cookies to identity the browser client and/or remember information such as 2FA login state. The core application makes use of these cookies:
+
+- `_open_project_session` (the name is configurable) contains the information about the logged in user as well as information stored between requests on the user's choices (e.g. the filters for costs are in part stored there)
+- `autologin` enables the user to automatically log in again after the session expired (e.g. because the browser was closed). It is set when the user checks the 'Stay logged in' box in the login form.
+- `op2fa_remember_token` the presence of that cookie suppresses the need for the user to provide a second factor upon login for 30 days if the user selects to do so when entering the 2fa information.
+
+On top of that, for cloud instances:
+
+- `openproject_cloud_instances` contains a list of instances the user recently accessed.
+- additional technical cookies might be set by the load balancer and intermediate processes (i.e., to remember a sticky load balancer and ensuring subsequent requests are routed similarly).
+
+
+
+# Processing of Personally Identifiable Information
+
+As OpenProject is dealing with user account data, it is processing personally identifiable information (PII). This data encompasses:
+
+- User's first and last name
+- E-Mail address(es)
+- Other uniquely identifiable information used as authentication data of external identity providers (such as SAML or OpenID Connect)
+- User pictures / Avatars
+- IP addresses in log files
```

Comment:
In case custom fields are used for users, that data might also fall into this category.